src/test/java/nextstep/subway/line/LineAcceptanceTest.java

```diff
@@ -1,85 +1,235 @@
+            }
+        );
     }
 
     @DisplayName("지하철 노선을 수정한다.")
     @Test
     void updateLine() {
         // given
         // 지하철_노선_등록되어_있음
+        ExtractableResponse<Response> createResponse = createTestLine1();
 
         // when
         // 지하철_노선_수정_요청
+        ExtractableResponse<Response> response = updateLine(getLocationId(createResponse), getTargetLine("3호선", "yellow lighten-1"));
 
         // then
         // 지하철_노선_수정됨
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.OK.value());
+    }
+
+    @DisplayName("지하철 노선을 수정할때, 기존에 존재하는 노선 이름으로는 변경할수 없다.")
+    @Test
+    void updateLine2() {
+        // given
+        // 지하철_노선_등록되어_있음
+        ExtractableResponse<Response> createResponse1 = createTestLine1();
+        ExtractableResponse<Response> createResponse2 = createTestLine2();
+
+        // when
+        // 지하철_노선_수정_요청
+        ExtractableResponse<Response> response = updateLine(getLocationId(createResponse1), getTargetLine("2호선", "green lighten-2"));
+
+        // then
+        // 지하철_노선_수정_실패
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.BAD_REQUEST.value());
     }
 
     @DisplayName("지하철 노선을 제거한다.")
     @Test
     void deleteLine() {
         // given
         // 지하철_노선_등록되어_있음
+        ExtractableResponse<Response> createResponse = createTestLine1();
 
         // when
         // 지하철_노선_제거_요청
+        long deletedId = getLocationId(createResponse);
+        ExtractableResponse<Response> deletedResponse = deleteLine(deletedId);
 
         // then
         // 지하철_노선_삭제됨
+        assertAll(
+            // 지하철_노선삭제
+            () -> assertThat(deletedResponse.statusCode()).isEqualTo(HttpStatus.NO_CONTENT.value()),
+
+            // 지하철_노선_찾지못함
+            () -> {
+                ExtractableResponse<Response> response = showLines(deletedId);
+                assertThat(response.statusCode()).isEqualTo(HttpStatus.NOT_FOUND.value());
+            }
+        );
+    }
+
+    private Map<String, String> getTargetLine(String name, String color) {
+        Map<String, String> params = new HashMap<String, String>(){
+            {
+                put("name", name);
+                put("color", color);
+            }
+        };
+        return Collections.unmodifiableMap(params);
+    }
+
+    private ExtractableResponse<Response> createTestLine1() {
+        return createLine(getTargetLine("1호선", "blue lighten-1"));
+    }
+
+    private ExtractableResponse<Response> createTestLine2() {
+        return createLine(getTargetLine("2호선", "green lighten-1"));
+    }
+
+    private ExtractableResponse<Response> createLine(Map<String, String> params) {
+        return RestAssured.given().log().all()
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .body(params)
+                .when()
+                .post(URI_PATH)
+                .then().log().all()
+                .extract();
+    }
```

Comment:
```suggestion
    
    private ExtractableResponse<Response> createTestLine1() {
        return createLine("1호선", "blue lighten-1");
    }

    private ExtractableResponse<Response> createTestLine2() {
        return createLine("2호선", "green lighten-1");
    }

    private ExtractableResponse<Response> createLine(String name, String color) {
        Map<String, String> params = new HashMap<String, String>(){
            {
                put("name", name);
                put("color", color);
            }
        };
        
        return RestAssured.given().log().all()
                .contentType(MediaType.APPLICATION_JSON_VALUE)
                .body(params)
                .when()
                .post(URI_PATH)
                .then().log().all()
                .extract();
    }
```

저는 위처럼 map 형태의 param을 작성하는 부분을 request 부분 내부로 이동하는 것을 제안드렸던 것인데요 
요청별로 param은 달라질 수 있기 때문에, 한 번 더 빼지 않은 부분입니다 
꼭 수정하셔야하는 부분은 아니니 참고만 해주세요 :)

Reply:
넵 다음커밋에 반영하겠습니다.